BraveWallet/Settings/CustomNetworkDetailsView.swift

```diff
@@ -0,0 +1,364 @@
+            } else {
+              item.error = Strings.Wallet.customNetworkInvalidAddressErrMsg
+              model.rpcUrls[index] = item
+            }
+          }
+        }
+      }
+      .listRowBackground(Color(.secondaryBraveGroupedBackground))
+      Section(
+        header:
+          Text(Strings.Wallet.customNetworkIconUrlsTitle)
+          .textCase(.none)
+      ) {
+        ForEach(model.iconUrls.indices, id: \.self) { index in
+          NetworkTextField(placeholder: Strings.Wallet.customNetworkUrlsPlaceholder,
+                           item: $model.iconUrls[index]) { newValue in
+            var item: NetworkInputItem = model.iconUrls[index]
+            if validateUrl(newValue) {
+              item.error = nil
+              model.iconUrls[index] = item
+              
+              if index == model.iconUrls.count - 1 { // add a new row
+                let newRow = NetworkInputItem(input: "")
+                model.iconUrls.append(newRow)
+              }
+            } else {
+              item.error = Strings.Wallet.customNetworkInvalidAddressErrMsg
+              model.iconUrls[index] = item
+            }
+          }
+        }
+      }
+      .listRowBackground(Color(.secondaryBraveGroupedBackground))
+      Section(
+        header:
+          Text(Strings.Wallet.customNetworkBlockExplorerUrlsTitle)
+          .textCase(.none)
+      ) {
+        ForEach(model.blockUrls.indices, id: \.self) { index in
+          NetworkTextField(placeholder: Strings.Wallet.customNetworkUrlsPlaceholder,
+                           item: $model.blockUrls[index]) { newValue in
+            var item: NetworkInputItem = model.blockUrls[index]
+            if validateUrl(newValue) {
+              item.error = nil
+              model.blockUrls[index] = item
+              
+              if index == model.blockUrls.count - 1 { // add a new row
+                let newRow = NetworkInputItem(input: "")
+                model.blockUrls.append(newRow)
+              }
+            } else {
+              item.error = Strings.Wallet.customNetworkInvalidAddressErrMsg
+              model.blockUrls[index] = item
+            }
+          }
+        }
+      }
+      .listRowBackground(Color(.secondaryBraveGroupedBackground))
+    }
+    .navigationBarTitle(Strings.Wallet.customNetworkDetailsTitle)
+    .navigationBarTitleDisplayMode(.inline)
+    .toolbar {
+      ToolbarItemGroup(placement: .confirmationAction) {
+        Button(action: {
+          addCustomNetwork()
+        }) {
+          Text(isEditMode ? Strings.Wallet.update : Strings.Wallet.add)
+            .foregroundColor(Color(.braveOrange))
+        }
+      }
+    }
+  }
+  
+  private func validateUrl(_ url: String) -> Bool {
+    let regex = "http[s]?://(([^/:.[:space:]]+(.[^/:.[:space:]]+)*)|([0-9](.[0-9]{3})))(:[0-9]+)?((/[^?#[:space:]]+)([^#[:space:]]+)?(#.+)?)?"
+    let test = NSPredicate(format: "SELF MATCHES %@", regex)
+    return test.evaluate(with: url)
+  }
```

Comment:
We should replace this and use `URIFixup.getURL` instead.  URIFixup though is in Client module and probably would need to be moved to BraveShared